geometry-processor.cpp

```diff
@@ -75,11 +75,13 @@ way_helper::~way_helper()
 }
 size_t way_helper::set(const osmid_t *node_ids, size_t node_count, const middle_query_t *mid)
 {
-    //if we don't have enough space already get more
-    if(node_cache.size() < node_count)
+    // other parts of the code assume that the node cache is the size of the way
+    // TODO: Fix this, and use std::vector everywhere
+    if(node_cache.size() != node_count)
         node_cache.resize(node_count);
-    //get the node data
+    // get the node data
     mid->nodes_get_list(&node_cache.front(), node_ids, node_count);
```

Comment:
Hmm, the return value of nodes_get_list() is ignored here, which is not good. The function returns the number of nodes actually copied in. If the geometry of some nodes is missing (think OSM data extract) the return value may be smaller than `node_count`. So there really is a second `resize()` necessary after the call that uses the return value or the vector still might be too large.